### ISSUE TITLE
v0.14.1 Release

### ISSUE THREAD

**Issue by crusaderky** [pydata/xarray#3434]
I think with the multiple recent breakages we've just had due to dependency upgrades, we should push out a patch release with some haste.

Please comment/add/object

Must have
--------------
- [x] numpy 1.18 support #3409
- [x] pseudonetcdf 3.1.0 support #3409, #3420 
- [x] require cftime != 1.0.4 #3463 
- [x] groupby reduce regression fix #3403
- [x] pandas master support #3440 
 
Nice to have
-----------------
- [x] ellipsis (...) work #1081, #3414, #3418, #3421, #3423, #3424
- [x] HTML repr #3425 (really mouth-watering, but I'm unsure about how far it is from completion)
- [x] groupby drop nan groups #3406 
- [x] deprecate `allow_lazy` #3435 
- [x] \_\_dask_tokenize\_\_ #3446 
- [x] dask name equality #3453
- [x] Leave empty slot when not using accessors #3531

**Comment by dcherian** [comment]
#3406 needs a review

**Comment by crusaderky** [comment]
@mathause I'm personally really uncomfortable pushing out a release when there are multiple known breakages against dependencies - see Must have list. Help is welcome!

**Comment by crusaderky** [comment]
@barronh any update re pseudonetcdf 3.1?

**Comment by barronh** [comment]
I believe there are two issues here.

First, the 3.1 release was not 2.7 compliant. It mixed named keyword arguments with `**kwds`. That could be fixed easily.

Second, the camx reader changed subclasses, which affects the xarray test for CF variables. I could update the test that would fix xarray test-case.

**Comment by crusaderky** [comment]
@barronh, we don't care about Python 2.7 compliance. We support >=3.6.

**Comment by barronh** [comment]
Got it. This is a test update to a test that is backward and forward compatible. I'll get something checked in.

**Comment by mathause** [comment]
xref for numpy 1.18 numpy/numpy#14831

**Comment by dcherian** [comment]
@mathause Thanks for filing these upstream issues!

**Comment by dcherian** [comment]
Once #3491 is merged, we should be ready to release since numpy will fix itself (https://github.com/numpy/numpy/pull/14841)

**Comment by mathause** [comment]
Don't we have to figure out whether to use `np.mean` or `np.nanmean` for datetime vectors?

**Comment by dcherian** [comment]
I'm in favour of releasing even if we are currently broken with numpy master. 

Our current release is also broken with numpy master, so the situation is no worse...

**Comment by crusaderky** [comment]
If someone could review and merge #3537 and #3531, we can close this!

**Comment by crusaderky** [comment]
I think we're good to go!

**Comment by dcherian** [comment]
I agree! @crusaderky, @max-sixty: do either of you have the time to make a release?

**Comment by max-sixty** [comment]
Yup I can do it later this afternoon!

**Comment by dcherian** [comment]
:+1:

**Comment by dcherian** [comment]
Thanks @max-sixty I activated the RTD build.
